From 48de011fa96a37c38265ae8ee2180f12e65f4de8 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:14 -0400
Subject: [tools] perf: Add missing -ldl for gtk build

Message-id: <1408462094-14194-146-git-send-email-jolsa@redhat.com>
Patchwork-id: 88097
O-Subject: [PATCH RHEL7.1 BZ1131394 145/325] perf tools: Add missing -ldl for gtk build
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit e2137086be7bc52893a790292635cfafc475b693
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Sep 26 20:55:54 2013 +0200

description
===========
If we build perf with NO_LIBPYTHON=1 NO_LIBPERL=1 the '-ldl' is not
added to libs build fails if we have gtk2 code in, because it depends on
it.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index cf6ad5d..29ad7d6 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -382,6 +382,7 @@ ifndef NO_GTK2
     CFLAGS += -DHAVE_GTK2_SUPPORT
     GTK_CFLAGS += $(shell pkg-config --cflags gtk+-2.0 2>/dev/null)
     GTK_LIBS := $(shell pkg-config --libs gtk+-2.0 2>/dev/null)
+    EXTLIBS += -ldl
   endif
 endif
 
-- 
1.7.1